op-node/eth/types.go

```diff
@@ -236,6 +238,33 @@ type PayloadAttributes struct {
 	GasLimit *Uint64Quantity `json:"gasLimit,omitempty"`
 }
 
+type L2BatchJustification struct {
+	PrevBatchLastBlock espresso.Header `json:"prevBatchLastBlock"`
+	FirstBlock espresso.Header `json:"firstBlock"`
+
+	// The block number of `FirstBlock`. This allows us to authenticate the `FirstBlock` header by
+	// computing the commitment of `FirstBlock` and checking it against the commitment at position
+	// `FirstBlockNumber` in a certified sequence (e.g. the sequencer contract). We can also
+	// authenticate `PrevBatchLastBlock` in a similar way, since it is required to be the block
+	// immediately before `FirstBlock`.
+	FirstBlockNumber uint64 `json:"firstBlockNumber"`
+
+	Payload *L2BatchPayloadJustification `json:"payload,omitempty",rlp:"nil"`
```

Comment:
Will the serialization panic without this rlp:"nil" tag?

Reply:
This affects deserialization. It makes it so that an empty RLP list deserializes as `nil`. Without this tag, the `rlp` library requires the data to represent a non-nil object or else deserialization fails. This field is specifically intended to be optional so we need to allow `nil`